File: Difficulty: Medium/Print Anagrams Together/print-anagrams-together.java
//{ Driver Code Starts
import java.util.*;


// } Driver Code Ends
class Solution {
    public ArrayList<ArrayList<String>> anagrams(String[] arr) {
        HashMap<String, ArrayList<String>> hm = new HashMap<>();
        
        for(String str : arr) {
            char chArr[] = str.toCharArray();
            Arrays.sort(chArr);
            String srtStr = new String(chArr);
            
            if(!hm.containsKey(srtStr))
                hm.put(srtStr, new ArrayList<>());
            
            hm.get(srtStr).add(str);
        }
        return new ArrayList<>(hm.values());
    }
}


//{ Driver Code Starts.

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        sc.nextLine(); // Ignore the newline after the test case input
        while (t-- > 0) {
            String inputLine = sc.nextLine();
            String[] arr = inputLine.split(" ");

            Solution ob = new Solution();
            ArrayList<ArrayList<String>> result = ob.anagrams(arr);
            result.sort(Comparator.comparing(a -> a.get(0)));
            for (ArrayList<String> group : result) {
                for (String word : group) {
                    System.out.print(word + " ");
                }
                System.out.println();
            }
            System.out.println("~");
        }
        sc.close();
    }
}

// } Driver Code Ends

File: Difficulty: Medium/Print Anagrams Together/print-anagrams-together.py
#User function Template for python3


class Solution:

    def anagrams(self, arr):
        '''
        words: list of word
        n:      no of words
        return : list of group of anagram {list will be sorted in driver code (not word in grp)}
        '''

        my_dict = {}
        
        for str in arr:
            chr_arr = list(str)
            chr_arr.sort()
            new_str = ''.join(chr_arr)
            
            if new_str not in my_dict:
                my_dict[new_str] = []
            
            my_dict[new_str].append(str)
            
        return list(my_dict.values())
#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__':
    t = int(input())
    for tcs in range(t):
        words = input().split()

        ob = Solution()
        ans = ob.anagrams(words)

        for grp in sorted(ans):
            for word in grp:
                print(word, end=' ')
            print()

        print("~")

# } Driver Code Ends